common/src/main/scala/com/gu/media/youtube/YouTubeClaims.scala

```diff
@@ -0,0 +1,126 @@
+import com.gu.media.logging.Logging
+import com.typesafe.config.Config
+import org.joda.time.DateTime
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+
+class YouTubeClaims(override val config: Config) extends YouTubeAccess with Logging {
+
+  private def createAsset(title: String, videoId: String): String = {
+
+    val metadata = new Metadata()
+      .setTitle(title)
+      .setDescription(videoId)
+
+    val asset = new Asset()
+      .setMetadata(metadata)
+      .setType("web")
+
+    val createdAsset = partnerClient.assets()
+      .insert(asset)
+      .setOnBehalfOfContentOwner(contentOwner)
+      .execute()
+
+    createdAsset.getId()
+
+  }
+
+  private def setOwnership(assetId: String) = {
+
+    val territoryOwners = new TerritoryOwners()
+      .setOwner(contentOwner)
+      .setRatio(100.0)
+      .setType("exclude")
+      .setTerritories(List[String]())
+
+    val ownership: RightsOwnership = new RightsOwnership().setGeneral(List(territoryOwners))
+
+    partnerClient.ownership()
+      .update(assetId, ownership)
+      .setOnBehalfOfContentOwner(contentOwner)
+      .execute()
+
+  }
+
+  private  def claimVideo(assetId: String, videoId: String, policy: Policy): Unit = {
+
+    val claim = new Claim()
+      .setAssetId((assetId))
+      .setVideoId(videoId)
+      .setPolicy(policy)
+      .setContentType("audiovisual")
+
+    val createdClaim = partnerClient.claims()
+      .insert(claim)
+      .setOnBehalfOfContentOwner(contentOwner)
+      .execute()
+  }
+
+  private def getPolicy(blockAds: Boolean): Policy = {
+    val policy = new Policy()
+    if (blockAds)
+      policy.setId(trackingPolicyId)
+    else
+      policy.setId(monetizationPolicyId)
+  }
+
+  private def createVideoClaim(atomId: String, userName: String, blockAds: Boolean, videoId: String): Unit = {
+    val policy = getPolicy(blockAds)
+    val assetTitle = s"$atomId-$userName-${DateTime.now()}"
+    val assetId = createAsset(assetTitle, videoId)
+    setOwnership(assetId)
+    claimVideo(assetId, videoId, policy)
+
+  }
+
+  private def updateClaim(claimId: String, assetId: String, videoId: String, blockAds: Boolean): Unit = {
+
+    val policy = getPolicy(blockAds)
+
+    val claim = new Claim()
+      .setAssetId((assetId))
+      .setVideoId(videoId)
+      .setPolicy(policy)
+      .setContentType("audiovisual")
+
+    partnerClient.claims
+      .patch(claimId, claim)
+      .setOnBehalfOfContentOwner(contentOwner)
+      .execute()
+  }
+
+  def createOrUpdateClaim(atomId: String, videoId: String, userName: String, blockAds: Boolean): Unit = {
+
+    try {
+      val videoClaims = partnerClient
+        .claimSearch()
+        .list
+        .setVideoId(videoId)
+        .execute()
+
+      if (videoClaims.getPageInfo.getTotalResults() == 0) {
+        createVideoClaim(atomId, userName, blockAds, videoId)
+      } else {
+
+        val claimsList = videoClaims.getItems().asScala
```

Comment:
> I didn't think it would be possible for one video to have so many claims

There's no limit to the number of claims a video can have on YouTube. Every copy of a video could get a claim. That is, if random people start ripping our videos and uploading them to their own channel, we would get a claim for each one. Obviously, the more people do this, the more claims. So we're probably making an assumption that our videos don't get ripped and uploaded, which may be ok for now...?